### PR TITLE
Release 2.0.0-alpha.1

### LINKED ISSUE #94: Add feedback system
Look into feedback system that Google products (taking screenshots/highlighting info/saving session data).

Probably this: https://github.com/ludrao/django-tellme

### PR DESCRIPTION
**Pull request by JackMorganNZ**
This is a massive PR, as `master` has not been updated in a long time. To view the new additions to the release branch since `develop`, see [this link](https://github.com/uccser/cs-unplugged/compare/develop...release/2.0.0-alpha.1).

Changes include:

- Add basic feedback form (fixes #94)
  - Form is a basic Google form. A fancier form can be created for `2.0.0` release.
- Increase visibility of unit plan link.
  - Now renders like topic and resource buttons.
- Remove duplicate CT links text from Binary Numbers junior lesson.
- Adds Computational Thinking page and introduction text before panels in each lesson/unit plan.
  - Also removes duplication of templates for CT sections.
- Fix link to changelog.
- Move changelog numbering explanation within note block.
- Increase visibility of links in footer by darkening text.
- Shrink jumbotron so small screens are prompted by presence of buttons.
  - For example, on a 15" Macbook, the buttons were barely visible to the user. By slightly shrinking the jumbotron the buttons are more visible, prompting the user to scroll to view these.

### CODE REVIEW

File: csunplugged/static/img/manifest.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        {
+            "src": "https://storage.googleapis.com/cs-unplugged-develop/static/img/android-chrome-256x256.png",
+            "sizes": "256x256",
+            "type": "image/png"
+        }
+    ],
+    "theme_color": "#ffffff",
+    "background_color": "#ffffff",
+    "display": "standalone"
+}
```
Comment (ravenmaster001) [inline csunplugged/static/img/manifest.json]: Is this abrupt EOF necessary?

File: csunplugged/tests/general/management/test_commands.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class ManagementCommandTest(BaseTestWithDB):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.language = "en"
+
+    def test_commands(self):
+        management.call_command("updatedata")
```
Comment (ravenmaster001) [inline csunplugged/tests/general/management/test_commands.py]: Should this really be tested in one test?

Comment (JackMorganNZ) [inline csunplugged/tests/general/management/test_commands.py]: Good call, I'll split these.

File: csunplugged/topics/urls.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        views.TopicView.as_view(),
+        name="topic"
+    ),
+    # eg: /topics/binary-numbers/integrations/binary-bracelets
+    url(
+        r"^(?P<topic_slug>[-\w]+)/integrations/(?P<integration_slug>[-\w]+)/$",
+        views.CurriculumIntegrationView.as_view(),
+        name="integration"
+    ),
+    # eg: /topics/binary-numbers/other-resources
```
Comment (hayleyavw) [inline csunplugged/topics/urls.py]: missing trailing `/`

File: csunplugged/topics/urls.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        views.AllCurriculumIntegrationList.as_view(),
+        name="all_curriculum_integrations"
+    ),
+    # eg: /topics/binary-numbers/
+    url(
+        r"^(?P<topic_slug>[-\w]+)/$",
+        views.TopicView.as_view(),
+        name="topic"
+    ),
+    # eg: /topics/binary-numbers/integrations/binary-bracelets
```
Comment (hayleyavw) [inline csunplugged/topics/urls.py]: missing trailing `/`

File: csunplugged/topics/management/commands/_UnitPlanLoader.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+class UnitPlanLoader(BaseLoader):
+    """Custom loader for loading unit plans."""
+
+    def __init__(self, factory, structure_file_path, topic, BASE_PATH):
+        """Create the loader for loading unit plans.
+
+        Args:
+            factory: LoaderFactory object for creating loaders (LoaderFactory).
+            structure_file_path: File path for structure YAML file (string).
+            topic: Object of related topic model.
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_UnitPlanLoader.py]: type?

File: csunplugged/topics/management/commands/_ProgrammingChallengesLoader.py [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+        self.BASE_PATH = os.path.join(self.BASE_PATH, os.path.split(structure_file_path)[0])
+        self.topic = topic
+
+    def load(self):
+        """Load the content for programming challenges.
+
+        Raise:
+            KeyNotFoundError: when no object can be found with the matching attribute.
+            MissingRequiredFieldError: when no object can be found with the matching
+                attribute.
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_ProgrammingChallengesLoader.py]: and `CouldNotFindMarkdownFileError`

File: csunplugged/topics/management/commands/_ProgrammingChallengesLoader.py [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+class ProgrammingChallengesLoader(BaseLoader):
+    """Custom loader for loading programming challenges."""
+
+    def __init__(self, structure_file_path, topic, BASE_PATH):
+        """Create the loader for loading programming challenges.
+
+        Args:
+            structure_file_path: File path for structure YAML file (string).
+            topic: Object of related topic model.
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_ProgrammingChallengesLoader.py]: type?

File: csunplugged/topics/management/commands/_LessonsLoader.py [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+
+class LessonsLoader(BaseLoader):
+    """Custom loader for loading lessons."""
+
+    def __init__(self, lessons_structure_file_path, topic, unit_plan, BASE_PATH):
+        """Create the loader for loading lessons.
+
+        Args:
+            lessons_structure_file_path: file path to lessons yaml file (str).
+            topic: Object of Topic model.
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_LessonsLoader.py]: type?

File: csunplugged/topics/management/commands/_LessonsLoader.py [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+        self.topic = topic
+        self.unit_plan = unit_plan
+
+    def load(self):
+        """Load the content for a single lesson.
+
+        Raises:
+            KeyNotFoundError: when no object can be found with the matching attribute.
+            MissingRequiredFieldError: when a value for a required model field cannot be
+                found in the config file.
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_LessonsLoader.py]: and `InvalidConfigValueError`

File: csunplugged/topics/management/commands/_LearningOutcomesLoader.py [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+
+class LearningOutcomesLoader(BaseLoader):
+    """Custom loader for loading learning outcomes."""
+
+    def __init__(self, structure_file_path, BASE_PATH):
+        """Create the loader for loading learning outcomes.
+
+        Args:
+            structure_file_path: File path to YAML file (string)
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_LearningOutcomesLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_LessonsLoader.py [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+    """Custom loader for loading lessons."""
+
+    def __init__(self, lessons_structure_file_path, topic, unit_plan, BASE_PATH):
+        """Create the loader for loading lessons.
+
+        Args:
+            lessons_structure_file_path: file path to lessons yaml file (str).
+            topic: Object of Topic model.
+            unit_plan: Object of UnitPlan model.
+            BASE_PATH: Base file path (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_LessonsLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_ProgrammingChallengesLoader.py [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+
+
+class ProgrammingChallengesLoader(BaseLoader):
+    """Custom loader for loading programming challenges."""
+
+    def __init__(self, structure_file_path, topic, BASE_PATH):
+        """Create the loader for loading programming challenges.
+
+        Args:
+            structure_file_path: File path for structure YAML file (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_ProgrammingChallengesLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_ProgrammingChallengesLoader.py [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+            topic: Object of related topic model.
+            BASE_PATH: Base file path (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_ProgrammingChallengesLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_ProgrammingChallengesStructureLoader.py [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+
+
+class ProgrammingChallengesStructureLoader(BaseLoader):
+    """Custom loader for loading structure of programming challenges."""
+
+    def __init__(self, structure_file_path, BASE_PATH):
+        """Create the loader for loading structure of programming challenges.
+
+        Args:
+            structure_file_path: File path for structure YAML file (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_ProgrammingChallengesStructureLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_ProgrammingChallengesStructureLoader.py [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+            BASE_PATH: Base file path (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_ProgrammingChallengesStructureLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_TopicLoader.py [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+
+class TopicLoader(BaseLoader):
+    """Custom loader for loading a topic."""
+
+    def __init__(self, factory, structure_file_path, BASE_PATH):
+        """Create the loader for loading a topic.
+
+        Args:
+            factory: LoaderFactory object for creating loaders (LoaderFactory).
+            structure_file_path: File path for structure YAML file (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_TopicLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_TopicLoader.py [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+            BASE_PATH: Base file path (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_TopicLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_UnitPlanLoader.py [hunk 18]
```diff
@@ -0,0 +1,159 @@
...
+
+class UnitPlanLoader(BaseLoader):
+    """Custom loader for loading unit plans."""
+
+    def __init__(self, factory, structure_file_path, topic, BASE_PATH):
+        """Create the loader for loading unit plans.
+
+        Args:
+            factory: LoaderFactory object for creating loaders (LoaderFactory).
+            structure_file_path: File path for structure YAML file (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_UnitPlanLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_UnitPlanLoader.py [hunk 19]
```diff
@@ -0,0 +1,159 @@
...
+            topic: Object of related topic model.
+            BASE_PATH: Base file path (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_UnitPlanLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_LearningOutcomesLoader.py [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+            BASE_PATH: Base file path (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_LearningOutcomesLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_GlossaryTermsLoader.py [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+
+
+class GlossaryTermsLoader(BaseLoader):
+    """Custom loader for loading glossary terms."""
+
+    def __init__(self, glossary_folder_path, structure_file_path, BASE_PATH):
+        """Create the loader for loading glossary terms.
+
+        Args:
+            glossary_folder_path: Folder path to definition files (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_GlossaryTermsLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_GlossaryTermsLoader.py [hunk 22]
```diff
@@ -0,0 +1,55 @@
...
+            structure_file_path: Path to the config file, used for errors.
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_GlossaryTermsLoader.py]: type?

File: csunplugged/topics/management/commands/_GlossaryTermsLoader.py [hunk 23]
```diff
@@ -0,0 +1,55 @@
...
+            BASE_PATH: Base file path (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_GlossaryTermsLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_CurriculumIntegrationsLoader.py [hunk 24]
```diff
@@ -0,0 +1,111 @@
...
+
+
+class CurriculumIntegrationsLoader(BaseLoader):
+    """Custom loader for loading curriculum integrations."""
+
+    def __init__(self, structure_file_path, topic, BASE_PATH):
+        """Create the loader for loading curriculum integrations.
+
+        Args:
+            structure_file_path: File path for structure YAML file (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_CurriculumIntegrationsLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_CurriculumIntegrationsLoader.py [hunk 25]
```diff
@@ -0,0 +1,111 @@
...
+            topic: Object of related topic model.
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_CurriculumIntegrationsLoader.py]: type?

File: csunplugged/topics/management/commands/_CurriculumIntegrationsLoader.py [hunk 26]
```diff
@@ -0,0 +1,111 @@
...
+            BASE_PATH: Base file path (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_CurriculumIntegrationsLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_CurriculumAreasLoader.py [hunk 27]
```diff
@@ -0,0 +1,121 @@
...
+
+
+class CurriculumAreasLoader(BaseLoader):
+    """Loader for curriculum area content."""
+
+    def __init__(self, structure_file_path, BASE_PATH):
+        """Create the loader for loading curriculum areas.
+
+        Args:
+            structure_file_path: File path to YAML file (string)
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_CurriculumAreasLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_CurriculumAreasLoader.py [hunk 28]
```diff
@@ -0,0 +1,121 @@
...
+            BASE_PATH: Base file path (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_CurriculumAreasLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_AgeGroupsLoader.py [hunk 29]
```diff
@@ -0,0 +1,79 @@
...
+
+
+class AgeGroupsLoader(BaseLoader):
+    """Loader for age group content."""
+
+    def __init__(self, structure_file_path, BASE_PATH):
+        """Create the loader for loading age groups.
+
+        Args:
+            structure_file_path: File path to YAML file (string)
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_AgeGroupsLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_AgeGroupsLoader.py [hunk 30]
```diff
@@ -0,0 +1,79 @@
...
+            BASE_PATH: Base file path (string).
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_AgeGroupsLoader.py]: `string` -> `str`

File: csunplugged/topics/management/commands/_AgeGroupsLoader.py [hunk 31]
```diff
@@ -0,0 +1,79 @@
...
+        self.structure_file_path = structure_file_path
+        self.BASE_PATH = os.path.join(self.BASE_PATH, os.path.split(structure_file_path)[0])
+
+    @transaction.atomic
+    def load(self):
+        """Load the content for age groups.
+
+        Raise:
+            MissingRequiredFieldError: when no object can be found with the
+                                       matching attribute.
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_AgeGroupsLoader.py]: random indentation here.

File: csunplugged/topics/content/en/sorting-networks/unit-plan/lessons/lessons.yaml [hunk 32]
```diff
@@ -0,0 +1,36 @@
...
+    sorting-network:
+      description: Use blank copy as template for creating Sorting Network.
+    sorting-network-cards:
+      description: Cards for students to use with the Sorting Network.
+  classroom-resources:
+    - 6 cards (or other items such as post-it notes) with numbers on them
+    - Stopwatch (e.g. on smartphone) to time races between groups.
+
+investigating-variations-using-the-sorting-network:
+  number: 2
```
Comment (hayleyavw) [inline csunplugged/topics/content/en/sorting-networks/unit-plan/lessons/lessons.yaml]: is this  meant to be here?

File: csunplugged/topics/content/en/error-detection-and-correction/curriculum-integrations/curriculum-integrations.yaml [hunk 33]
```diff
@@ -0,0 +1,36 @@
...
+    unit-plan:
+      - parity-magic
+
+word-study-activity:
+  number: 5
+  curriculum-areas:
+    - literacy
+  prerequisite-lessons:
+    unit-plan:
+      - parity-magic
```
Comment (hayleyavw) [inline csunplugged/topics/content/en/error-detection-and-correction/curriculum-integrations/curriculum-integrations.yaml]: EOF

File: csunplugged/topics/content/en/binary-numbers/unit-plan/lessons/lessons.yaml [hunk 34]
```diff
@@ -0,0 +1,115 @@
+how-binary-digits-work-junior:
+  number: 1
```
Comment (hayleyavw) [inline csunplugged/topics/content/en/binary-numbers/unit-plan/lessons/lessons.yaml]: remove number

File: csunplugged/topics/content/en/binary-numbers/unit-plan/lessons/lessons.yaml [hunk 35]
```diff
@@ -0,0 +1,115 @@
...
+  generated-resources:
+    binary-cards:
+      description: One set for class demonstration.
+    binary-windows:
+      description: One set per student.
+  classroom-resources:
+    - Number line from 0 to 20
+
+reinforcing-sequencing-in-binary-number-systems-junior:
+  number: 2
```
Comment (hayleyavw) [inline csunplugged/topics/content/en/binary-numbers/unit-plan/lessons/lessons.yaml]: remove number

File: csunplugged/templates/topics/topic-other-resources.html [hunk 36]
```diff
@@ -0,0 +1,25 @@
...
+  {% breadcrumb topic.name "topics:topic" topic.slug %}
+  {% breadcrumb "Other Resources" "topics:other_resources" topic.slug %}
+{% endblock breadcrumbs %}
+
+{% block page_heading %}
+  <h1>{{ topic.name }}</h1>
+{% endblock page_heading %}
+
+{% block left_column_content %}
+  <h2>Other Resources</h2>
```
Comment (hayleyavw) [inline csunplugged/templates/topics/topic-other-resources.html]: should be lowercase `r`

File: csunplugged/templates/topics/programming-challenge-language-solution.html [hunk 37]
```diff
@@ -0,0 +1,63 @@
...
+{% endblock breadcrumbs %}
+
+{% block page_heading %}
+  <h1>
+    {{ programming_challenge.challenge_set_number }}.{{ programming_challenge.challenge_number }}
+    {{ programming_challenge.name }}
+  </h1>
+  <h2>
+    <img src="{% get_static_prefix %}{{ implementation.language.icon }}" class="inline-image" />
+    {{ implementation.language.name }} Solution
```
Comment (hayleyavw) [inline csunplugged/templates/topics/programming-challenge-language-solution.html]: should be lowercase `s`

File: csunplugged/templates/resources/base-resource-pdf.html [hunk 38]
```diff
@@ -0,0 +1,38 @@
...
+  </header>
+  {% for resource_image_data in resource_images %}
+    {% for resource_image in resource_image_data %}
+      <div class="base-image-container">
+        <img class="base-image" src="data:image/png;base64,{{ resource_image }}" />
+      </div>
+    {% endfor %}
+  {% endfor %}
+  <footer class="fixed-running-element">
+    {% load static %}
```
Comment (hayleyavw) [inline csunplugged/templates/resources/base-resource-pdf.html]: why is this here and not at the top of the file?

File: csunplugged/resources/views/treasure_hunt.py [hunk 39]
```diff
@@ -0,0 +1,133 @@
...
+
+    Used after the resource name in the filename, and
+    also on the resource image.
+
+    Args:
+        request: HTTP request object
+        resource: Object of resource data.
+
+    Returns:
+        text for subtitle (string)
```
Comment (hayleyavw) [inline csunplugged/resources/views/treasure_hunt.py]: `string` - >`str`

File: csunplugged/resources/views/sorting_network_cards.py [hunk 40]
```diff
@@ -0,0 +1,136 @@
...
+
+    Used after the resource name in the filename, and
+    also on the resource image.
+
+    Args:
+        request: HTTP request object
+        resource: Object of resource data.
+
+    Returns:
+        text for subtitle (string)
```
Comment (hayleyavw) [inline csunplugged/resources/views/sorting_network_cards.py]: `string` -> `str`

File: csunplugged/resources/views/sorting_network.py [hunk 41]
```diff
@@ -0,0 +1,113 @@
...
+
+    Used after the resource name in the filename, and
+    also on the resource image.
+
+    Args:
+        request: HTTP request object
+        resource: Object of resource data.
+
+    Returns:
+        text for subtitle (string)
```
Comment (hayleyavw) [inline csunplugged/resources/views/sorting_network.py]: `string` -> `str`

File: csunplugged/resources/views/parity_cards.py [hunk 42]
```diff
@@ -0,0 +1,85 @@
...
+
+    Used after the resource name in the filename, and
+    also on the resource image.
+
+    Args:
+        request: HTTP request object
+        resource: Object of resource data.
+
+    Returns:
+        text for subtitle (string)
```
Comment (hayleyavw) [inline csunplugged/resources/views/parity_cards.py]: `string` -> `str`

File: csunplugged/resources/views/generate_resource_pdf.py [hunk 43]
```diff
@@ -0,0 +1,95 @@
...
+MM_TO_PIXEL_RATIO = 3.78
+
+
+def generate_resource_pdf(request, resource, module_path):
+    """Return a response containing a generated PDF resource.
+
+    Args:
+        request: HTTP request object
+        resource: Object of resource data.
+        module_path: Path to module for generating resource.
```
Comment (hayleyavw) [inline csunplugged/resources/views/generate_resource_pdf.py]: string?

File: csunplugged/resources/views/binary_to_alphabet.py [hunk 44]
```diff
@@ -0,0 +1,116 @@
...
+
+    Used after the resource name in the filename, and
+    also on the resource image.
+
+    Args:
+        request: HTTP request object
+        resource: Object of resource data.
+
+    Returns:
+        text for subtitle (string)
```
Comment (hayleyavw) [inline csunplugged/resources/views/binary_to_alphabet.py]: `string` -> `str`

File: csunplugged/resources/views/binary_cards_small.py [hunk 45]
```diff
@@ -0,0 +1,114 @@
...
+
+    Used after the resource name in the filename, and
+    also on the resource image.
+
+    Args:
+        request: HTTP request object
+        resource: Object of resource data.
+
+    Returns:
+        text for subtitle (string)
```
Comment (hayleyavw) [inline csunplugged/resources/views/binary_cards_small.py]: `string` -> `str`

File: csunplugged/resources/views/binary_cards.py [hunk 46]
```diff
@@ -0,0 +1,113 @@
...
+
+    Used after the resource name in the filename, and
+    also on the resource image.
+
+    Args:
+        request: HTTP request object
+        resource: Object of resource data.
+
+    Returns:
+        text for subtitle (string)
```
Comment (hayleyavw) [inline csunplugged/resources/views/binary_cards.py]: `string` -> `str`

File: csunplugged/resources/models.py [hunk 47]
```diff
@@ -0,0 +1,23 @@
...
+    webpage_template = models.CharField(max_length=200)
+    generation_view = models.CharField(max_length=200)
+    thumbnail_static_path = models.CharField(max_length=200)
+    copies = models.BooleanField()
+
+    def __str__(self):
+        """Text representation of Resource object.
+
+        Returns:
+            Name of resource (string).
```
Comment (hayleyavw) [inline csunplugged/resources/models.py]: `string` -> `str`

File: csunplugged/resources/management/commands/_ResourcesLoader.py [hunk 48]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class ResourcesLoader(BaseLoader):
+    """Custom loader for loading resources."""
+
+    def __init__(self, structure_file, BASE_PATH):
+        """Create the loader for loading resources.
+
+        Args:
+            structure_file: file path for structure YAML file (string)
```
Comment (hayleyavw) [inline csunplugged/resources/management/commands/_ResourcesLoader.py]: `string` -> `str`

File: csunplugged/resources/management/commands/_ResourcesLoader.py [hunk 49]
```diff
@@ -0,0 +1,60 @@
...
+            BASE_PATH: base file path (string)
```
Comment (hayleyavw) [inline csunplugged/resources/management/commands/_ResourcesLoader.py]: `string` -> `str`

File: csunplugged/general/urls.py [hunk 50]
```diff
@@ -0,0 +1,14 @@
...
+
+from . import views
+
+urlpatterns = [
+    url(r"^about/$", views.GeneralAboutView.as_view(), name="about"),
+    url(r"^contact/$", views.GeneralContactView.as_view(), name="contact"),
+    url(r"^people/$", views.GeneralPeopleView.as_view(), name="people"),
+    url(r"^principles/$", views.GeneralPrinciplesView.as_view(), name="principles"),
+    url(r"^computational-thinking/$", views.ComputationalThinkingView.as_view(), name="computational_thinking"),
+    url(r"^$", views.GeneralIndexView.as_view(), name="home"),
```
Comment (hayleyavw) [inline csunplugged/general/urls.py]: layout of these urls doesn't match the resource or topic url files

File: csunplugged/topics/management/commands/_AgeGroupsLoader.py (hunk 29 above)
Comment (ravenmaster001) [inline csunplugged/topics/management/commands/_AgeGroupsLoader.py]: what's wrong with string? (except that str is more explicit typing)

File: csunplugged/utils/BaseLoader.py [hunk 51]
```diff
@@ -0,0 +1,159 @@
...
+            mdx_math.MathExtension()
+        ]
+        self.converter = Verto(html_templates=templates, extensions=extensions)
+
+    def convert_md_file(self, md_file_path, config_file_path, heading_required=True, remove_title=True):
+        """Return the Verto object for a given Markdown file.
+
+        Args:
+            md_file_path: Location of Markdown file to convert (str).
+            config_file_path: Path to related the config file (str).
```
Comment (ravenmaster001) [inline csunplugged/utils/BaseLoader.py]: Yes, where are the optional args?

File: infrastructure/app-sample.yaml [hunk 52]
```diff
@@ -0,0 +1,12 @@
+runtime: custom
```
Comment (ravenmaster001) [inline infrastructure/app-sample.yaml]: Should we even keep this file any more?

File: Dockerfile [hunk 53]
```diff
@@ -0,0 +1,30 @@
...
+EXPOSE 8080
+RUN mkdir /csunplugged
+WORKDIR /csunplugged
+
+# Copy and install Python dependencies
+RUN python -m virtualenv --python=python3.4 /docker_venv
+COPY requirements /requirements
+RUN /docker_venv/bin/pip3 install -r /requirements/production.txt
+
+ADD ./csunplugged /csunplugged/
```
Comment (ravenmaster001) [inline Dockerfile]: Should we login with a no-root user i.e. see render service dockerfile in my WIP PR

File: csunplugged/templates/resources/base-resource-pdf.html (hunk 38 above)
Comment (JackMorganNZ) [inline csunplugged/templates/resources/base-resource-pdf.html]: Because it doesn't have to be, I put it next to the one place it is used, however for consistency I've placed it at the top.

File: infrastructure/app-sample.yaml (hunk 52 above)
Comment (JackMorganNZ) [inline infrastructure/app-sample.yaml]: It was more a starter point for someone else wanting to deploy this to GAE.

**Comment by JackMorganNZ** [comment]
Thanks for that @hayleyavw, Google image was temporarily unavailable so I've restarted the build.
